lib/warren/queue_broadcast_message.rb

```diff
@@ -1,15 +1,28 @@
 class Warren::QueueBroadcastMessage
+  include AfterCommitEverywhere
+
   attr_reader :record
 
-  def initialize(record)
-    @record = record
+  def initialize(record = nil, class_name: nil, id: nil)
+    if record
+      @record = record
+      @class_name = record.class.name
+      @id = record.id
+    else
+      @class_name = class_name
+      @id = id
+    end
+  end
+
+  def queue(warren)
+    after_commit { warren << self }
```

Comment:
I can see this is used in plate.rb, but not quite getting how it works and why it was not needed before? Thanks

Reply:
Previously the after save callback on well would use the after_commit on well to queue the well object for broadcast. However that depends on us instantiating and persisting the entire wells for the duration of the transaction, when all we need is the class name and the id. This lets us create a mini message consisting of just the class name and id, and adding that to the transaction instead.

I've actually popped a story in the backlog to look at doing this everywhere, as it should let us substantially reduce the memory requirements of very large transactions. (Aliquots in particular can get VERY expensive)

In this particular case the performance saving wasn't quite as great as I hoped when I started (I must have messed up on my 'quick and dirty version' as it looked like things got 10 times faster again. But I couldn't reproduce that even when I stopped creating wells completely) but left it in due to the memory saving.